README.md

```diff
@@ -1,106 +1,193 @@
 
-[What is the end-user need which this project aims to address?]
+The second goal of this API is to abstract away the network stack, allowing applications that need to access key-value stores to not worry about what network protocol is used to access the key-value store.
 
 ### Non-goals
 
-[If there are "adjacent" goals which may appear to be in scope but aren't, enumerate them here. This section may be fleshed out as your design progresses and you encounter necessary technical and other trade-offs.]
+- Cover all application requirements for a key-value store. Instead, it focuses on the most common use cases. It allows providers to extend the API to support more use cases.
+- Transactional semantics.
+- Data consistency.
+- Data replication.
+- Data sharding.
+- Cluster management.
+- Monitoring
 
 ### API walk-through
 
-[Walk through of how someone would use this API.]
+#### Use case 1
 
-#### [Use case 1]
+Imagine you have an HTTP handler that needs to persist some data to a key-value store. The handler needs to be able to retrieve, delete, and update the data in the key-value store. The following Rust code shows how you can use the WASI Key-Value Store API to in the handler.
 
-[Provide example code snippets and diagrams explaining how the API would be used to solve the given problem]
+```rust
+let value = b"value1";
+kv1::create_or_update("key", value)?;
+let val = kv1::get("key")?;
+assert_eq!(val, value);
 
-#### [Use case 2]
-
-[etc.]
+kv1::delete("key")?;
+```
 
 ### Detailed design discussion
 
-[This section should mostly refer to the .wit.md file that specifies the API. This section is for any discussion of the choices made in the API which don't make sense to document in the spec file itself.]
+```go
+interface "wasi:kv/data/crud" {
+  use { Error, payload, key } from "wasi:kv/types"
+
+  get: func(key: key) -> result<payload, Error>
+
+  create-or-update: func(key: key, value: stream<u8>) -> result<_, Error>
 
-#### [Tricky design choice #1]
+  delete: func(key: key) -> result<_, Error>
+}
 
-[Talk through the tradeoffs in coming to the specific design point you want to make.]
+interface "wasi:kv/data/increment" {
+  use { Error, key } from "wasi:kv/types"
+  
+  increment: func(key: key) -> result<u64, Error>
+}
 
+interface "wasi:kv/data/bulk-get" {
+  use { Error, payload } from "wasi:kv/types"
+  
+  bulk-get: func(keys: keys) -> result<stream<payload>, Error>
+
+  keys: func() -> result<keys, Error>
+}
+
+interface "wasi:kv/data/bulk-put" {
+  use { Error, key } from "wasi:kv/types"
+  
+  bulk-create-or-update: func(key_values: stream<(key, stream<u8>)>) -> result<_, Error>
+}
+
+interface "wasi:kv/data/ttl" {
```

Comment:
I'd rather see ttl as an option on put in the basic interface than a separate interface for just ttl.

Reply:
I agree with you on that an extra interface for ttl seems redundent. The concern for adding ttl to the basic operation interface is that it makes the interface less general and forces users to either opt-out the ttl parameter by providing a `None` value or not.

Reply:
I've moved the ttl interface to "under discssion" section.